src/Microsoft.DncEng.SecretManager/Readme.md

```diff
@@ -0,0 +1,196 @@
+# The .NET Core Engineering Secret Manager
+This tool is used to track and rotate the secrets in one or more key vaults, and to validate the usages of the same. This is done through a yaml manifest file for each vault, and json settings files for each service.
+
+## Rotating secrets
+The synchronize command can be used to rotate secrets that need rotation. It will automatically rotate all secrets that have either expired or hit their rotation time. This command is run as part of a scheduled build to keep all secrets up to date. Any secrets that can be rotated without a human involved will be rotated automatically. For secrets that require human interaction, the build will fail and a human will have to run synchronize.
+
+## Creating new secrets
+New secrets can be created by adding new entries to the manifest file, then running synchronize. The tool will then create the new secret and add it to the secret store.
+
+## Manifest Files
+A manifest file describes the expected state of a single secret store (currently only azure key vault, but support for other stores can be added easily). The manifest contains an inventory of all secrets and the information required to generate/rotate them. The format is as follows.
+```yaml
+# The destination secret store. All the secrets described in this document will be placed in here.
+storageLocation:
+  type: azure-key-vault
+  parameters:
+    name: <a string>
+    subscription: <a guid>
+
+# A collection of named references to other secret stores, secrets in this document can reference values from these stores if they are required
+references:
+  reference1:
+    type: azure-key-vault
+    parameters:
+      name: <a string>
+      subscription: <a guid>
+
+
+# encryption keys that are needed in the vault, these keys will be created if they don't exist.
+keys:
+  encryption-key-one:
+    type: RSA
+    size: 2048
+
+# each entry below is a separate secret that will be maintained in the secret store
+# the type identifies what kind of secret it is, and the parameters drive the creation of the secret
+secrets:
+  secret1:
```

Comment:
I think this example would be more useful if it matched a real world situation. Maybe something from our own patterns of use.

Reply:
"SecretReference" is a good example of this as I really don't understand how it is meant to be used. 

What is a "raw string" in this context? Is it the name of a secret that is defined elsewhere in the YAML?

Reply:
The synchronize command will either fail, or prompt the user if a TTY is detected. I can add that.

Reply:
I can do that.

Reply:
"raw string" just means "you passed a single string, instead of an object"

Reply:
I updated the examples. Does it make sense now?